### PR TITLE
virtio_scsi_cdrom: Add an installation case

### PR DESCRIPTION
**Pull request by YongxueHong**
Add a new case that guest could install os from
scsi-cd device for x86_64.

ID: 1628043
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/1960

**Comment by YongxueHong** [comment]
@luckyh  @CongLi  @lijinlijin  @peixiu  please help me review this patch, thanks.

File: qemu/tests/cfg/virtio_scsi_cdrom.cfg [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    # don't have enough RAM error from anaconda, please enlarge this value.
+    lowest_mem = 512
+    only x86_64
+    only virtio_scsi, virtio_blk
+    virtio_drive_letter = 'D'
+    variants:
+        - with_installation:
+            cd_format_cd1 = scsi-cd
+            i440fx:
+                cd_format_unattended = ide
```
Comment (CongLi) [inline qemu/tests/cfg/virtio_scsi_cdrom.cfg]: should be only for windows guest.

Comment (YongxueHong) [reply qemu/tests/cfg/virtio_scsi_cdrom.cfg]:
```
            i440fx:
                cd_format_unattended = scsi-cd
                Windows:
                    cd_format_unattended = ide
```
Thanks for your comment, have updated.

File: qemu/tests/cfg/virtio_scsi_cdrom.cfg [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    only x86_64
+    only virtio_scsi, virtio_blk
+    virtio_drive_letter = 'D'
+    variants:
+        - with_installation:
+            cd_format_cd1 = scsi-cd
+            i440fx:
+                cd_format_unattended = scsi-cd
+                Windows:
+                    cd_format_unattended = ide
```
Comment (CongLi) [inline qemu/tests/cfg/virtio_scsi_cdrom.cfg]: cd_format = scsi-cd
        - with_installation:
            i440fx:
                Windows:
                    cd_format_unattended = ide
                q35:
                    cd_format_unattended = ahci

Comment (YongxueHong) [reply qemu/tests/cfg/virtio_scsi_cdrom.cfg]: Thanks for your comment, have updated.

**Comment by YongxueHong** [comment]
@CongLi @lijinlijin @peixiu please help me review this patch, thanks.

**Comment by peixiu** [comment]
Hi yhong, 
It seems  virtio_scsi_cdrom.cfg only cover one case unattended_install.scsi_cdrom,  if this case can be  written in unattended_install.cfg as a variants -scsi_cdrom? like:
    variants:
        # Install guest from scsi cdrom
        - scsi_cdrom:
                cd_format_cd1 = scsi-cd

**Comment by YongxueHong** [comment]
Hi, peixiu
This just to cover a case installing os from scsi_cd, while the main purpose of unattended_install is to prepare a guest os for the next case.
The detail reason for adding a new case for this scenario, not add a variant in unattended_install.cfg, you could ask about @CongLi and @luckyh , we have talked about it. thanks.

**Comment by YongxueHong** [comment]
@CongLi @lijinlijin @peixiu please help me review this patch, thanks.

**Comment by CongLi** [comment]
Ack.

**Comment by peixiu** [comment]
02:08:46 INFO | PASS 1-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_scsi_cdrom.with_installation.cdrom.extra_cdrom_ks.aio_native.q35

Thanks, that's ok for me~
Acked-by: Peixiu phou@redhat.com

**Comment by YongxueHong** [comment]
@hereischen help me to review it , thanks

File: qemu/tests/cfg/virtio_scsi_cdrom.cfg [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    force_create_image = yes
+    guest_port_unattended_install = 12323
+    kernel = vmlinuz
+    initrd = initrd.img
+    # Throw errors if guest screen is inactive
+    inactivity_watcher = error
+    # Inactivity treshold to error the test
+    inactivity_treshold = 1800
+    image_verify_bootable = no
+    # Backup images from nfs when install failed
```
Comment (hereischen) [inline qemu/tests/cfg/virtio_scsi_cdrom.cfg]: we don't have the nfs for this funtion any more.

Comment (YongxueHong) [reply qemu/tests/cfg/virtio_scsi_cdrom.cfg]: Thanks, have updated.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by YongxueHong** [comment]
Thanks all.

**Pull request merged**